src/mero_elasticache.erl

```diff
@@ -77,17 +77,29 @@ request_response(Host, Port, Command, Names) ->
     case gen_tcp:connect(Host, Port, Opts) of
         {ok, Socket} ->
             ok = gen_tcp:send(Socket, Command),
-            Lines = [{Name, begin
-                                {ok, Line} = gen_tcp:recv(Socket, 0, 10000),
-                                Line
-                            end}
-                || Name <- Names],
+            Lines = receive_lines(Names, Socket),
             ok = gen_tcp:close(Socket),
```

Comment:
Maybe its better to also trap exceptions here, like…
```erlang
try receive_lines(Names, Socket)
catch
    _:Error ->
        %% Log the error and the stacktrace
        {error, Error}
after
  ok = gen_tcp:close(Socket)
end
```

Reply:
And if you do so you can throw an error from `receive_lines/3` instead of returning it.